File: java/com/countrygamer/angel/client/ClientProxy.java
package com.countrygamer.angel.client;

import net.minecraft.client.Minecraft;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.world.World;
import net.minecraftforge.common.MinecraftForge;

import com.countrygamer.angel.client.gui.GuiVortex;
import com.countrygamer.angel.client.gui.HUDOverlay;
import com.countrygamer.angel.client.render.RenderWeepingAngel;
import com.countrygamer.angel.client.render.RenderWeepingAngelStatue;
import com.countrygamer.angel.client.render.TileEntityPlinthRenderer;
import com.countrygamer.angel.common.CommonProxy;
import com.countrygamer.angel.common.entity.EntityStatue;
import com.countrygamer.angel.common.entity.EntityWeepingAngel;
import com.countrygamer.angel.common.tile.TileEntityPlinth;

import cpw.mods.fml.client.registry.ClientRegistry;
import cpw.mods.fml.client.registry.RenderingRegistry;

public class ClientProxy extends CommonProxy {
	
	@Override
	public void registerRender() {
		MinecraftForge.EVENT_BUS.register(new HUDOverlay(Minecraft.getMinecraft()));
		
		RenderingRegistry.registerEntityRenderingHandler(EntityWeepingAngel.class,
				new RenderWeepingAngel(0.5f));
		RenderingRegistry.registerEntityRenderingHandler(EntityStatue.class,
				new RenderWeepingAngelStatue());
		
		ClientRegistry.bindTileEntitySpecialRenderer(TileEntityPlinth.class,
				new TileEntityPlinthRenderer());
		
	}
	
	@Override
	public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
		if (ID == 0) return new GuiVortex(player);
		return null;
	}
	
}

File: src/main/scala/com/countrygamer/weepingangels/client/ClientProxy.scala
package com.countrygamer.weepingangels.client

import java.util

import com.countrygamer.weepingangels.client.gui.configFactory.GuiConfig
import com.countrygamer.weepingangels.client.gui.{GuiStatue, GuiTimeManipulation, HUDOverlay}
import com.countrygamer.weepingangels.client.render.{RenderWeepingAngel, TERendererStatue}
import com.countrygamer.weepingangels.common.entity.{EntityAngelArrow, EntityWeepingAngel}
import com.countrygamer.weepingangels.common.tile.TileEntityStatue
import com.countrygamer.weepingangels.common.{CommonProxy, WAOptions}
import cpw.mods.fml.client.IModGuiFactory
import cpw.mods.fml.client.IModGuiFactory.{RuntimeOptionCategoryElement, RuntimeOptionGuiHandler}
import cpw.mods.fml.client.registry.{ClientRegistry, RenderingRegistry}
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiScreen
import net.minecraft.client.renderer.entity.RenderArrow
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.tileentity.TileEntity
import net.minecraft.util.Vec3
import net.minecraft.world.World
import net.minecraftforge.common.MinecraftForge

/**
 *
 *
 * @author CountryGamer
 */
class ClientProxy() extends CommonProxy with IModGuiFactory {

	override def registerRender(): Unit = {

		ClientRegistry.bindTileEntitySpecialRenderer(
			classOf[TileEntityStatue], new TERendererStatue()
		)

		RenderingRegistry.registerEntityRenderingHandler(
			classOf[EntityWeepingAngel], new RenderWeepingAngel()
		)

		RenderingRegistry.registerEntityRenderingHandler(
			classOf[EntityAngelArrow], new RenderArrow()
		)

		MinecraftForge.EVENT_BUS.register(HUDOverlay)

	}

	override def getClientElement(ID: Int, player: EntityPlayer, world: World, coord: Vec3,
			tileEntity: TileEntity): AnyRef = {
		if (ID == WAOptions.statueGui && tileEntity.isInstanceOf[TileEntityStatue]) {
			return new GuiStatue(tileEntity.asInstanceOf[TileEntityStatue])
		}
		else if (ID == WAOptions.timeManipGui) {
			return new GuiTimeManipulation(player)
		}
		null
	}

	override def initialize(minecraftInstance: Minecraft): Unit = {

	}

	override def runtimeGuiCategories(): util.Set[RuntimeOptionCategoryElement] = {
		null
	}

	override def getHandlerFor(element: RuntimeOptionCategoryElement): RuntimeOptionGuiHandler = {
		null
	}

	override def mainConfigGuiClass(): Class[_ <: GuiScreen] = {
		classOf[GuiConfig]
	}

}
